egs/wsj/s5/steps/libs/nnet3/train/chain_objf/acoustic_model.py

```diff
@@ -20,32 +21,64 @@
 logger.addHandler(logging.NullHandler())
 
 
-def create_phone_lm(dir, tree_dir, run_opts, lm_opts=None):
-    """Create a phone LM for chain training
+def create_phone_lm(dir, tree_dir, run_opts, alignment_dirs=None, lm_opts=None):
```

Comment:
Vimal pointed out that this might fail due to argument list too long.  But you can work around that by doing:
``` for x in [arg_list]; do gunzip -c $x; done | ... ```
I don't like the C++ code changes because having a scale that is required to be an integer is kind of weird, and because IMO it's just more elegant to repeat the input that many times, since that is what the code does.  And also because of the principle that the "unusual case" shouldn't cause code to be modified in things used  in the "common case".  I want to keep the common case simple, which makes the codebase easier to maintain.